pilot/pkg/leaderelection/leaderelection.go

```diff
@@ -60,7 +64,9 @@ type LeaderElection struct {
 
 	// Criteria to determine leader priority.
 	revision       string
+	remote         bool
 	prioritized    bool
+	ignoreLocation bool // turn on for testing with old key comparision
```

Comment:
Sorry for the back and forth.. I don't think we should have the old code in the prod code path, since we introduce a lot of complex there.

I was thinking in the prod code we have
`config.KeyComparison = LocationPrioritizedComaprision // not an inline function now`

Then in the tests we have:

```go
compFunctions := []func(){
  LocationPrioritizedComaprision,
  func() { ... old logic ...}
}

Then we can just add simple unit tests for these and not need to do the full (somewhat heavy) leader election end-to-end flow we have now. We can make a table driven test for all cases

Reply:
> I was thinking in the prod code we have
>  config.KeyComparison = LocationPrioritizedComaprision // not an inline function now

The KeyComparison function just takes a single string argument, but our implementation also needs access to the both the config and the leaderelection struct, so it needs to be written in-scope, IIUC.

Reply:
```
config.KeyComparison = func(s string) {
  LocationPrioritizedComaprision(s, someOtherDep)
}
```
should work

Reply:
I split out the compare function and made the test simpler, table driven.